packages/engine/src/datastore/table/proxy.rs

```diff
@@ -237,17 +234,17 @@ impl
 }
 
 impl StateWriteProxy {
-    pub fn new<K: WriteState>(state: &mut K) -> Result<Self> {
+    pub fn new(state: &mut State) -> Result<Self> {
```

Comment:
This isn't a critique of the PR, but a question about our general design.
Do the proxies enforce that `Self` has the same lifetime as `&mut` here?

Reply:
No, the proxies may outlive `Self` as they share the ownership.

It would be nice if this could be enforced somehow, but the current design doesn't allow this as we pass the proxies by messages and can't track how long they live.

Reply:
No, proxies can't have limited lifetimes, because then they couldn't be sent between threads. But I agree that we should pass `State` (or `Box<State>`) by value where possible, instead of using write proxies.